gen/namespace.go

```diff
@@ -76,9 +91,27 @@ func (n *namespace) NewName(base string) string {
 		name = fmt.Sprintf("%s%d", base, i)
 	}
 	n.taken[name] = struct{}{}
+
+	if _, ok := n.gave[base]; !ok {
```

Comment:
since maps return default values (`nil` for slice) and `nil` is a valid slice, you can just do
```go
n.gave[base] = append(n.gave[base], name)
```
no need for the check here.